baseframe/forms/validators.py

```diff
@@ -238,7 +238,10 @@ def check_url(self, invalid_urls, url, text=None):
                     # For text patterns, do a substring search. For regex patterns (assumed so if not text),
                     # do a regex search. Test with the final URL from the response, after redirects,
                     # but report errors using the URL the user provided
-                    if (isinstance(pattern, basestring) and pattern in rurl) or (pattern.search(rurl) is not None):
+                    if isinstance(pattern, basestring):
+                        if pattern in rurl:
+                            return message.format(url=url, text=text)
+                    elif pattern.search(rurl) is not None:
                         return message.format(url=url, text=text)
```

Comment:
The other syntax works, but this seems a little clearer. Would you prefer the other syntax?